src/main/scala/mesosphere/marathon/core/leadership/impl/AbdicateOnConnectionLossActor.scala

```diff
@@ -0,0 +1,40 @@
+package mesosphere.marathon.core.leadership.impl
+
+import akka.actor.{ Actor, ActorLogging, Props }
+import com.twitter.common.zookeeper.ZooKeeperClient
+import mesosphere.marathon.LeadershipAbdication
+import org.apache.zookeeper.{ WatchedEvent, Watcher }
+
+object AbdicateOnConnectionLossActor {
+  def props(zk: ZooKeeperClient, leader: LeadershipAbdication): Props = {
+    Props(new AbdicateOnConnectionLossActor(zk, leader))
+  }
+}
+
+class AbdicateOnConnectionLossActor(zk: ZooKeeperClient,
+                                    leader: LeadershipAbdication) extends Actor with ActorLogging {
+
+  private[impl] val watcher = new Watcher {
+    val reference = self
+    override def process(event: WatchedEvent): Unit = reference ! event
+  }
+
+  override def preStart(): Unit = {
+    log.info("Register as ZK Listener")
+    zk.register(watcher)
+  }
+  override def postStop(): Unit = {
+    log.info("Unregister as ZK Listener")
+    zk.unregister(watcher)
+  }
+
+  def disconnected(): Unit = {
+    log.warning("ZooKeeper connection has been dropped. Abdicate Leadership.")
+    leader.abdicateLeadership()
+  }
+
+  override def receive: Receive = {
+    case event: WatchedEvent if event.getState == Watcher.Event.KeeperState.Disconnected => disconnected()
```

Comment:
That appears to be rather specific. I would rather like to test for a healthy state. Other than that we might want to deal with `Expired` as well?

These are possible states:

```java
        public enum KeeperState {
            /** Unused, this state is never generated by the server */
            @Deprecated
            Unknown (-1),

            /** The client is in the disconnected state - it is not connected
             * to any server in the ensemble. */
            Disconnected (0),

            /** Unused, this state is never generated by the server */
            @Deprecated
            NoSyncConnected (1),

            /** The client is in the connected state - it is connected
             * to a server in the ensemble (one of the servers specified
             * in the host connection parameter during ZooKeeper client
             * creation). */
            SyncConnected (3),

            /**
             * Auth failed state
             */
            AuthFailed (4),

            /**
             * The client is connected to a read-only server, that is the
             * server which is not currently connected to the majority.
             * The only operations allowed after receiving this state is
             * read operations.
             * This state is generated for read-only clients only since
             * read/write clients aren't allowed to connect to r/o servers.
             */
            ConnectedReadOnly (5),

            /**
              * SaslAuthenticated: used to notify clients that they are SASL-authenticated,
              * so that they can perform Zookeeper actions with their SASL-authorized permissions.
              */
            SaslAuthenticated(6),

            /** The serving cluster has expired this session. The ZooKeeper
             * client connection (the session) is no longer valid. You must
             * create a new client connection (instantiate a new ZooKeeper
             * instance) if you with to access the ensemble. */
            Expired (-112);
           // ...
}
```